### DISCUSSION TITLE
commitMessageTopic config option ignored on branch updates

### DISCUSSION THREAD

**Discussion by gizero** [renovatebot/renovate#17631] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

it was likely v32.159.0 at the time

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab CE v15.2.3

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

With the config bits for the `kustomize` manager attached below, which works as expected for new branches/MRs, I noticed Renovate ignores the `commitMessageTopic` value on branch updates. To me it seems it's switching back to the option default value while updating a branch content.

```json
  "kustomize": {
    "additionalBranchPrefix": "{{parentDir}}-",
    "commitMessageTopic": "dependency {{depName}} in {{parentDir}}"
  }
```

The unexpected behaviour is best explained by the MR events history from Gitlab:

<img width="1006" alt="Screenshot 2022-09-05 at 11 49 41" src="https://user-images.githubusercontent.com/1544064/188424271-e0acf058-c2f1-4c45-b56f-d06d35835678.png">

As you can see, the initial commit message was `chore(deps): update dependency traefik in testing to v2.8.2` but was changed to `chore(deps): update traefik docker tag to v2.8.3`, yet losing the `"in {{parentDir}}"` template bits, when the next image update was detected.

Instead, the `additionalBranchPrefix` value is preserved upon updates, as expected.
Unfortunately I don't have any Renovate logs for that run, since the relevant artefacts are now expired and they have been deleted already.

I'm a fairly new adopter of the `commitMessageTopic` option to separate MRs for different application environments, then I have a too low mileage to tell whether this used to work in the past for branch updates as well.

I'll keep an eye on next updates and will try to collect Renovate logs if/when this happens again. Please tell me how can I otherwise help to provide more context and debug this issue.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
this happen because of a recent bug fix. the docker datasource is overriding the `commitMessageTopic`, so you need to use a package rule instead of the `kustomize` object

**Reply by viceice** [reply]
This is a known limitation and isn't solvable yet. This is also not a bug, so pleaser open a discussion first in future.

**Reply by gizero** [reply]
@viceice thanks for the quick feedback.
> this happen because of a recent bug fix.

Sounds like "something got fixed" and, as a side-effect, we  Am I reading it correctly? If so, is there a relevant issue/MR you can share? I couldn't easily find any. I hope I can gain a better insight on the topic/limitation?

Sorry for the improper workflow but, to me, this looked like a "buggy" behaviour ;-) . That said, next time I'll start a discussion first.

**Reply by viceice** [reply]
- #17102
